typescript/cli/src/config/agent.ts

```diff
@@ -67,6 +70,16 @@ export async function createAgentConfig({
     startBlocks,
   );
 
+  try {
+    AgentConfigSchema.parse(agentConfig);
+  } catch (e) {
+    const error = fromError(e);
+    logRed(
+      `Agent config is invalid, this is possibly due to required contracts not being deployed. See details below:\n${error.toString()}`,
```

Comment:
Using `zod-validation-error` to show users a better error message, seems to be maintained.

An example error message

```
yarn hyperlane registry agent-config --chains anvil8545
Hyperlane CLI

Creating agent config...
Agent config is invalid, this is possibly due to required contracts not being deployed. See details below
Validation error: Required at "chains.anvil8545.interchainGasPaymaster"
```